src/providers/walletconnectNotify.ts

```diff
@@ -0,0 +1,161 @@
+    const subscribers: string[] = await subscribersRs.json();
+
+    return subscribers;
+  } catch (e) {
+    capture('[WalletConnect] failed to fetch subscribers');
+    return [];
+  }
+}
+
+// Find the CAIP10 of subscribers, since the Notify API requires CAIP10.
+async function crossReferenceSubscribers(space: { id: string }, spaceSubscribers) {
+  const subscribersFromDb = spaceSubscribers;
+  const subscribersFromWalletConnect = await getSubscribersFromWalletConnect();
+
+  // optimistically reserve all subscribers from the db
+  const crossReferencedSubscribers = new Array(subscribersFromDb.length);
+
+  // Create a hashmap for faster lookup
+  const addressPrefixMap = new Map<string, string>();
+  for (const subscriber of subscribersFromWalletConnect) {
+    const unprefixedAddress = subscriber.split(':').pop();
+    if (unprefixedAddress) {
+      addressPrefixMap.set(unprefixedAddress, subscriber);
+    }
+  }
+
+  for (const subscriber of subscribersFromDb) {
+    const crossReferencedAddress = addressPrefixMap.get(subscriber);
+    if (crossReferencedAddress) {
+      crossReferencedSubscribers.push(crossReferencedAddress);
+    }
+  }
+
+  // remove empty elements from the array, since some might not have been found in WalletConnect Notify server
+  return crossReferencedSubscribers.filter(addresses => addresses);
+}
+
+async function queueNotificationsToSend(notification, accounts: string[]) {
+  for (let i = 0; i < accounts.length; i += MAX_ACCOUNTS_PER_REQUEST) {
+    await sendNotification(notification, accounts.slice(i, i + MAX_ACCOUNTS_PER_REQUEST));
+    const waitTime = 1 / PER_SECOND_RATE_LIMIT + WAIT_ERROR_MARGIN;
+    await wait(waitTime);
+  }
+}
+
+export async function sendNotification(notification, accounts) {
+  const notifyUrl = `${WALLETCONNECT_NOTIFY_SERVER_URL}/${WALLETCONNECT_PROJECT_ID}/notify`;
+
+  const body = {
+    accounts,
+    notification
+  };
+
+  try {
+    const notifyRs = await fetch(notifyUrl, {
+      method: 'POST',
+      headers: {
+        ...AUTH_HEADER,
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(body)
+    });
+
+    const notifySuccess = await notifyRs.json();
+
+    return notifySuccess;
+  } catch (e) {
+    capture('[WalletConnect] failed to notify subscribers', e);
+  }
+}
+
+// Transform proposal event into notification format.
+async function formatMessage(event, proposal) {
+  const space = proposal.space;
+  if (!space) return null;
+
+  const notificationType = getNotificationType(event.event);
+  const notificationBody = getNotificationBody(event.event, space);
+
+  if (!notificationType) {
+    capture(`[WalletConnect] could not get matching notification type for event ${event.event}`);
+    return;
+  }
+
+  if (!notificationBody) {
+    capture(`[WalletConnect] could not get matching notification body for event ${event.event}`);
+    return;
+  }
+
+  const url = new URL(proposal.link);
+  url.searchParams.append('app', 'walletconnect');
```

Comment:
This is resulting in `https://snapshot.org/?app=walletconnect#/thanku.eth/proposal/0x8c2a67398c9912f7c2ba15a68caf94388ea98a88074079103d48e3eae83e2ea7` I think it is better to add string at the end of URL

Reply:
What's the purpose of `?app=walletconnect`?

Reason for `new URL` was to properly parse and construct the string. What would the string at the end of the URL be used for?

Reply:
> What would the string at the end of the URL be used for?

It is used by snapshot to know that someone voted from web3inbox, changed it to `web3inbox` in the other PR, hope it is fine